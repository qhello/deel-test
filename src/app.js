import express from "express";
import bodyParser from "body-parser";

import { sequelize } from "./model.js";
import { getProfile } from "./middleware/getProfile.js";
import controllers from "./controllers/index.js";

const app = express();
app.use(bodyParser.json());
app.set("sequelize", sequelize);
app.set("models", sequelize.models);

/**
 * @returns contract by id, only if it belongs to the profile calling
 */
app.get("/contracts/:id", getProfile, controllers.getContractById);

/**
 * @returns contracts that belongs to the profile calling
 */
app.get("/contracts", getProfile, controllers.getContracts);

/**
 * @returns jobs, that are unpaid & belongs to contracts that belongs to the profile calling
 */
app.get("/jobs/unpaid", getProfile, controllers.getJobsUnpaid);

/**
 * @returns 200 OK, or error. Will attempt to pay an unpaid job
 */
app.post("/jobs/:job_id/pay", getProfile, controllers.postJobsPay);

/**
 * @returns 200 OK, or error. Will attempt to deposit specified money into client's balance
 */
app.post(
  "/balances/deposit/:userId",
  getProfile,
  controllers.postBalanceDeposit
);

app.get("/admin/best-profession", controllers.getBestProfession);

app.get("/admin/best-clients", controllers.getBestClients);

export default app;
