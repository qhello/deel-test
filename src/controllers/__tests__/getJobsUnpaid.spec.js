import request from "supertest";
import app from "../../app.js";

describe("getJobsUnpaid", () => {
  it("should return jobs", async () => {
    const res = await request(app).get("/jobs/unpaid").set("profile_id", 2);

    expect(res.statusCode).toEqual(200);
    expect(res.body).toMatchSnapshot([
      {
        Contract: {
          createdAt: expect.any(String),
          updatedAt: expect.any(String),
        },
        createdAt: expect.any(String),
        updatedAt: expect.any(String),
      },
      {
        Contract: {
          createdAt: expect.any(String),
          updatedAt: expect.any(String),
        },
        createdAt: expect.any(String),
        updatedAt: expect.any(String),
      },
    ]);
  });
});
