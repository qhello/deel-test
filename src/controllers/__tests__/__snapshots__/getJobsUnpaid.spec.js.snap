// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getJobsUnpaid should return jobs 1`] = `
[
  {
    "Contract": {
      "ClientId": 2,
      "ContractorId": 6,
      "createdAt": Any<String>,
      "id": 3,
      "status": "in_progress",
      "terms": "bla bla bla",
      "updatedAt": Any<String>,
    },
    "ContractId": 3,
    "createdAt": Any<String>,
    "description": "work",
    "id": 3,
    "paid": null,
    "paymentDate": null,
    "price": 202,
    "updatedAt": Any<String>,
  },
  {
    "Contract": {
      "ClientId": 2,
      "ContractorId": 7,
      "createdAt": Any<String>,
      "id": 4,
      "status": "in_progress",
      "terms": "bla bla bla",
      "updatedAt": Any<String>,
    },
    "ContractId": 4,
    "createdAt": Any<String>,
    "description": "work",
    "id": 4,
    "paid": null,
    "paymentDate": null,
    "price": 200,
    "updatedAt": Any<String>,
  },
]
`;
